### PR TITLE
[MID-24] add text input field component

### PR DESCRIPTION
**Pull request by IvanKoshan**
Added a text input field. 
Ticket: https://trello.com/c/K7WqiYvW/24-%D0%B7%D0%B2%D0%BE%D1%80%D0%BE%D1%82%D0%BD%D1%96%D0%B9-%D0%B7%D0%B2%D1%8F%D0%B7%D0%BE%D0%BA-%D0%BA%D0%BE%D0%BC%D0%BF%D0%BE%D0%BD%D0%B5%D0%BD%D1%82-%D0%BF%D0%BE%D0%BB%D1%8F-%D0%B2%D0%B2%D0%BE%D0%B4%D1%83

demo: https://ivan-koshans-mayak.vercel.app/test/textinputfield

### CODE REVIEW

File: src/app/test/textinputfield/page.js
Comment (dmitrenko-v) [inline src/app/test/textinputfield/page.js:1]: 🟠 Do we need this file?

**Review by dmitrenko-v — APPROVED** [review]

File: src/app/_components/Icons/InpurErrorIcon.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export default function InputErrorIcon({ className }) {
```
Comment (alex-popov-tech) [inline src/app/_components/Icons/InpurErrorIcon.js]: 🟠 please prefer named export

**Review by alex-popov-tech — CHANGES_REQUESTED** [review]

File: src/app/_components/InputFields/TextInputField.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      >
+        <input
+          className={cn(
+            variant.input.base,
+            variant.input.focus,
+            variant.input.style,
+            error && variant.input.error,
+          )}
+          id={id}
+          type="text"
```
Comment (yataras) [inline src/app/_components/InputFields/TextInputField.js]: @IvanKoshan Consider making this component more flexible to cover also other than `text` types. Eg. `email`, `password`, `url`. You can pass `type` as a property and define needed list of possible values in the `propTypes`. "text" could be a defaultProps value.

File: src/app/_components/Icons/InpurErrorIcon.js
Comment (yataras) [inline src/app/_components/Icons/InpurErrorIcon.js:1]: @IvanKoshan A typo in the file name. Please change `InpurErrorIcon` -> `InputErrorIcon`

**Review by yataras — CHANGES_REQUESTED** [review]

File: src/app/_components/Icons/InpurErrorIcon.js (hunk 1 above)
Comment (IvanKoshan) [reply src/app/_components/Icons/InpurErrorIcon.js]: fixed

File: src/app/_components/InputFields/TextInputField.js (hunk 2 above)
Comment (IvanKoshan) [reply src/app/_components/InputFields/TextInputField.js]: did it

File: src/app/_components/Icons/InpurErrorIcon.js
Comment (IvanKoshan) [reply src/app/_components/Icons/InpurErrorIcon.js:1]: fixed it

**Review by yataras — APPROVED** [review]

**Review by alex-popov-tech — APPROVED** [review]

**Pull request merged**